File: United_Community/Amit/Daily-Leetcode/day_10_valid_parentheses.cpp
class Solution {
public:
    bool isValid(std::string s) {
        int in = 0;
        
        for (char ch : s) {
            if (ch == '[' || ch == '(' || ch == '{') {
                s[in++] = ch;
            } else if (in == 0) {
                return false;
            } else if ((ch == ']' && s[in - 1] == '[') || 
                       (ch == '}' && s[in - 1] == '{') || 
                       (ch == ')' && s[in - 1] == '(')) {
                in--;
            } else {
                return false;
            }
        }
        
        return in == 0;
    }
};

File: United_Community/Mradul/Daily-Leetcode/day_10_ValidParentheses.java
// https://leetcode.com/problems/valid-parentheses/

class Solution {
    public boolean isValid(String s) {
        char[] ch = s.toCharArray();
        int in = 0;
        for (int i = 0; i < ch.length; i++) {
            if (ch[i] == '[' || ch[i] == '(' || ch[i] == '{') {
                ch[in++] = ch[i];
            }
            else if  (in == 0) {
                return false;
            }
            else if ((ch[i] == ']' && ch[in-1] == '[') || (ch[i] == '}' && ch[in-1] == '{') || (ch[i] == ')' && ch[in-1] == '(')) {
                in--;
            } 
            else {
                return false;
            }
        }
        return in==0;
    }
}